update PhoneNumberFormatter

contargo-development opened the pull request:
currently by some numbers the last digits where deleted.
because non german numbers have more spaces in the areacode and connection number
so if they are splitted by the array has a greater length.

daschJulia committed:
update PhoneNumberFormatter

currently by some numbers the last digits where deleted.
because non german numbers have more spaces in the areacode and connection number
so if they are splitted by the array has a greater length.

olle commented:
Are there tests at all that capture the throwing behaviour?

rjayasinghe commented on src/main/java/net/contargo/types/telephony/formatting/PhoneNumberFormatter.java:
```diff
@@ -216,26 +225,29 @@ private AreaCodeAndConnectionNumber getAreaCodeAndConnectionNumberFromFixedNumbe
             throw new PhoneNumberFormattingException(String.format("Could not extract anything from input number: %s",
                     preFormattedNumber));
         } else if (splittedPreformattedNumber.length <= 2) { // two elements -> take the second element as connection
```
Not sure if that really applies to reality - but what if there are more than 3 elements e.g. more spaces in the number. Imagine a number within a company where you could have [countrycode] [city code] [connection number] [extension]. Normally that extension would be seperated by a '-'. But since the task of this lib is formatting malformed numbers we should expect something like this.

To cut a long story short. I'd have code that takes all remaining elements and hand them over as VarArgs to the NewNationalNumber constructor.

rjayasinghe replied:
OK. With some support I read the code again and it handles exactly the described case. ;)

rjayasinghe reviewed (approved):
looks good to me. @contargo-development please merge.

daschJulia committed:
change validator and add empty constructor for PhoneNumbers

change constructor to be able to get empty Objects.
let the validator work directly on PhoneNumber objects instead of
creating temporary ones.

daschJulia committed:
change Validator

now the vaalidator report contraint violations in phonenumber
there are 3 different types.

rjayasinghe commented on src/main/java/net/contargo/types/telephony/validation/PhoneNumberValidationResult.java:
```diff
@@ -0,0 +1,8 @@
+
+    ZERO_NUMBER,
+    CAN_NOT_FORMATTED,
+    TO_LARGE
```
TOO

rjayasinghe reviewed (approved):
Go for it! :)

daschJulia committed:
use correct propertyName and remove typo

in other apps are the propertyNames mobile or phone.

daschJulia committed:
prepare release
